### ISSUE TITLE
makefile cleanups?

### ISSUE THREAD

**Issue by sagiegurari** [sunshine-protocol/sunshine-bounty-ui#12]
Really cool project. However, i think the makefile there is hard to maintain due to all the duplication.
i'll give an example:

```toml
[tasks.android-armv7-release]
private = true
condition = { env_true = ["RELEASE"] }
command = "cargo"
args = [
  "ndk", 
  "--platform", 
  "${ANDROID_PLATFORM_VERSION}", 
  "--target", 
  "armv7-linux-androideabi", 
  "build",
  "--release"
  ]

[tasks.android-i686-release]
private = true
condition = { env_true = ["RELEASE"] }
command = "cargo"
args = [
  "ndk", 
  "--platform", 
  "${ANDROID_PLATFORM_VERSION}", 
  "--target", 
  "i686-linux-android", 
  "build",
  "--release"
  ]
```

These 2 are the same except the target value.
there are a lot of ways to solve this. one way is this:

```toml
[tasks.android-release]
private = true
condition = { env_true = ["RELEASE"], env_set = ["ANDROID_TARGET"] }
command = "cargo"
args = [
  "ndk", 
  "--platform", 
  "${ANDROID_PLATFORM_VERSION}", 
  "--target", 
  "${ANDROID_TARGET}", 
  "build",
  "--release"
]

[tasks.android-armv7-release]
private = true
env = { ANDROID_TARGET = "armv7-linux-androideabi" }
run_task = "android-release"

[tasks.android-i686-release]
private = true
env = { ANDROID_TARGET = "i686-linux-android" }
run_task = "android-release"
```

**Comment by shekohex** [comment]
Thank you very much, I already got to know the `run_task` option 2 or 3 days ago:

https://github.com/sunshine-protocol/sunshine-identity-ui/blob/9d74f6890dc8b8eacf3c89bc3b3ace60e9f6941e/Makefile.toml#L304-L314

when I got an idea to change the `crate-type` on the fly since cargo does not support this yet.
For example, iOS only need `staticlib`, and android only needs `cdynlib`.
with cargo make that was easy, we edit Cargo.toml file with the `create-type` before doing anything, and clean everything at the end.

I'm going to improve the other scripts too.
